Minio Role Added

Bharat-DRS opened the pull request.

Bharat-DRS committed:
Minio Role Added

dzdimov commented on vagrant/provisioning/arkcase-all.yml:
```diff
@@ -1,6 +1,8 @@
 - hosts: all
   roles:
+    - role: minio
+      tags: [minio]
```
Let's put this down under microservices role

dzdimov commented on vagrant/provisioning/arkcase-ce.yml:
```diff
@@ -1,6 +1,8 @@
 - hosts: localhost
   roles:
+    - role: minio
+      tags: [minio]
```
Let's put this down under microservices role

dzdimov commented on vagrant/provisioning/arkcase-dev.yml:
```diff
@@ -1,6 +1,8 @@
 - hosts: localhost
   roles: 
+    - role: minio
+      tags: [minio]
```
Let's put this down under microservices role

dzdimov commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,58 @@
+  group:
+    name: minio
+    state: present
+- name: create minio user
```
please spaces between each tasks.
This goes for all the other tasks down this main.yml

dzdimov commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,58 @@
+- name: create minio bin directory
+  become: yes
+  file:
+    path: "{{root_folder}}/minio/{{item}}"
```
you are missing the /app directory {{ root_folder }}/app/minio/{{ item }}

dzdimov replied:
Also please leave spaces when defining variables {{ variable }} instead of {{variable}}

dzdimov commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,58 @@
+- name: check if minio file exists
+  become: yes
+  stat: 
+   path: "{{root_folder}}/minio/bin/minio"
```
you are missing the /app directory {{ root_folder }}/app/minio/{{ item }}

dzdimov commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,58 @@
+  become: yes
+  get_url:
+    url: "https://dl.minio.io/server/minio/release/linux-amd64/minio"
+    dest: "{{root_folder}}/minio/bin/"
```
you are missing the /app directory {{ root_folder }}/app/minio/{{ item }}

dzdimov commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,58 @@
+- name: create minio config file
+  become: yes
+  copy:
+    dest: "{{root_folder}}/minio/minio.conf"
```
you are missing the /app directory {{ root_folder }}/app/minio/{{ item }}

Bharat-DRS committed:
Modified Minio from /minio to /app/minio and added to microservices

bojmil-armedia commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+  copy:
+    dest: "{{root_folder}}/app/minio/minio.conf"
+    content: |
+     MINIO_VOLUMES= "{{root_folder}}/app/minio/data"
```
The data should go in {{root_folder}}/data/minio/

Bharat-DRS replied:
sure, will make changes as suggested

bojmil-armedia commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+  become: yes
+  copy:
+    dest: "{{root_folder}}/app/minio/minio.conf"
+    content: |
```
Can you add something for logging? Logs should go to {{root_folder}}/log/minio/

Bharat-DRS replied:
The application logs are going to system console

bojmil-armedia commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+    owner: minio
+  with_items:
+    - "bin"
+    - "data"
```
No need for data directory here. Create this directory in  {{root_folder}}/data/minio/

bojmil-armedia replied:
That being said you don't really need bin directory, because {{root_folder}}/app/minio/ should be the bin directory.

Bharat-DRS replied:
> No need for data directory here. Create this directory in {{root_folder}}/data/minio/
All the created Volumes are going to be located to  **{{root_folder}}/app/minio/data**  so Minio need a Data Directory, we can move this to **{{root_folder}}/data/minio/**, will make changes.


> That being said you don't really need bin directory, because {{root_folder}}/app/minio/ should be the bin directory.
The Minio Executable file is located in **{{root_folder}}/app/minio/bin**

Bharat-DRS replied:
changed the folder structure as per the suggestion

david-oc-miller commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+- name: check if minio file exists
+  become: yes
+  stat: 
+   path: "{{root_folder}}/app/minio/bin/minio"
```
It has to follow the standard folder structure.

`{{ root_folder }}/app/minio`, `{{ root_folder }}/data/minio`, `{{ root_folder }}/log/minio`, `{{ root_folder }}/tmp/minio`.

david-oc-miller replied:
Most likely the logging configuration needs to be modified, to ensure the log files go into `{{ root_folder }}/log/minio`.

Bharat-DRS replied:
sure, will make changes as per the suggested structure.

Bharat-DRS replied:
made changes ads per the suggestion

david-oc-miller commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+    - "bin"
+    - "data"
+
+- name: check if minio file exists
```
The download task in common module checks whether the file exists and downloads if necessary.  Use that task, instead of these two tasks (`stat` and `get_url`)

Bharat-DRS replied:
Sure, we will merge this

david-oc-miller replied:
@Bharat-DRS , don't forget to switch to the shared download task.

Bharat-DRS replied:
Changed the download task and now using the Common task to download the minio file

david-oc-miller replied:
@Bharat-DRS I see you are still using `get_url`, and not using the download task at all.  The download task would look like this;

```
- include_tasks: "{{ role_path }}/../common/tasks/download.yml"
  loop:
    - name: "MariaDB JDBC Driver"
      dest: "{{ root_folder }}/common/mariadb-java-client-{{ mariadb_jdbc_versio
n }}.jar"
      owner: "root"
      url: "https://downloads.mariadb.com/Connectors/java/connector-java-{{ mari
adb_jdbc_version }}/mariadb-java-client-{{ mariadb_jdbc_version }}.jar"
```

This is an example from the mariadb-client role.

david-oc-miller commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+    masked: no
+    daemon_reload: yes
+
+- name: start minio
```
MinIO has to listen on TLSv1.2 ; for FedRAMP we require all services to be TLS-enabled.

david-oc-miller commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,66 @@
+  get_url:
+    url: "https://dl.minio.io/server/minio/release/linux-amd64/minio"
+    dest: "{{root_folder}}/app/minio/bin/"
+    mode: '0777'
```
For Red Hat 8 support, we also need to set `setype: bin_t` on the executable file.

Bharat-DRS replied:
sure, we will add this

Bharat-DRS replied:
added the setype to the downloading file

bojmil-armedia commented:
@Bharat-DRS please address the comments, so we can merge this one.

Bharat-DRS committed:
new changes

created /app /data /tmp /log folders and changed the file destinations accordingly

Bharat-DRS committed:
updated minio role

download task modified, changed to use Common download task

david-oc-miller commented on vagrant/provisioning/roles/minio/tasks/main.yml:
```diff
@@ -0,0 +1,73 @@
+    - "log"
+    - "tmp"
+
+- name: download minio file from internet
```
Ah this looks very nice now :-) the only problem is the `download.yml` doesn't support `bin_t`.  Can you now set `bin_t` mode in your `file` task, below this one?  Then it should be good to go.

Bharat-DRS committed:
updated minio role

Moved the setype to the file permission task
